feat(managed_kubernetes): Add customizing cilium configuration guide (#8797)
ovh/docs: pages/public_cloud/containers_orchestration/managed_kubernetes/customizing-cilium/guide.en-gb.md

updated: 2025-12-11
---

## Objective

The OVHcloud Managed Kubernetes service provides you with Kubernetes clusters without the hassle of installing or operating them.

The Standard plan of OVHcloud Managed Kubernetes clusters uses [Cilium](https://cilium.io/) as the default CNI.

The Cilium agent process (a.k.a. DaemonSet) allows configuration on a per-node basis.

This allows overriding cilium-config ConfigMap for a node or set of nodes by using `CiliumNodeConfig` objects.

> [!warning]
> Without using a `CiliumNodeConfig` object, it will not be possible to update cilium-config ConfigMap.

## Requirements

- An OVHcloud Managed Kubernetes cluster with Standard plan.

## Instructions

### What is CiliumNodeConfig

As stated in [Cilium documentation](https://docs.cilium.io/en/stable/configuration/per-node-config/#ciliumnodeconfig-objects):

A `CiliumNodeConfig` object allows for overriding ConfigMap / Agent arguments.

It consists of a set of fields and a label selector. The label selector defines to which nodes the configuration applies.

As is the standard with Kubernetes, an empty LabelSelector (e.g. `{}`) selects all nodes.

### CiliumNodeConfig possible values

> You can retrieve all keys and values in the cilium-configmap file of the [Cilium github repository](https://github.com/cilium/cilium/blob/main/install/kubernetes/cilium/templates/cilium-configmap.yaml).

Reviewer: ```suggestion
You can find the full list of possible keys and values in the [cilium-configmap file on github](https://github.com/cilium/cilium/blob/main/install/kubernetes/cilium/templates/cilium-configmap.yaml).
```

Reviewer 2: Suggestion applied, thanks @pgaxatte!

> [!warning]

> > Be aware that some keys may require enabling certain features in the Cilium operator, which are disabled by default.

Reviewer: ```suggestion
> Be aware that some keys may require enabling certain features in the Cilium operator that might be enabled on our platform.
```

Reviewer 2: Modified by:
> Be aware that some keys available on our platform may require certain features to be enabled in the Cilium operator.
